data/write_data.rb

```diff
@@ -0,0 +1,56 @@
+def save_persons
+  store_person = @persons.map do |person, _index, _id|
+    if person.instance_of?(Teacher)
+      {
+        occupation: 'Teacher',
+        name: person.name,
+        age: person.age,
+        specialization: person.specialization
+      }
+    else
+      {
+        occupation: 'Student',
+        name: person.name,
+        age: person.age,
+        parent_permission: person.parent_permission
+      }
```

Comment:
- [ ] Kindly note that you need to save also the classroom of the student so that you may not get errors when trying to load your preserved data.

Reply:
✅ RESOLVED: We have ensured that the classroom of the student is saved.